src/future_feature_flags.h

```diff
@@ -0,0 +1,63 @@
+// Flags to enable upcoming features
+#ifndef FISH_FUTURE_FEATURE_FLAGS_H
+#define FISH_FUTURE_FEATURE_FLAGS_H
+
+#include <assert.h>
+
+class features_t {
+public:
+    /// The list of flags.
+    enum flag_t {
+        /// Whether ^ is supported for stderr redirection.
+        stderr_nocaret,
+
+        /// The number of flags.
+        flag_count
+    };
+
+    /// Return whether a flag is set.
+    bool test(flag_t f) const {
+        assert(f >= 0 && f < flag_count && "Invalid flag");
+        return values[f];
+    }
+
+    /// Set a flag.
+    void set(flag_t f, bool value) {
+        assert(f >= 0 && f < flag_count && "Invalid flag");
+        values[f] = value;
+    }
+
+    /// Metadata about feature flags.
+    struct metadata_t {
+        /// The flag itself.
+        features_t::flag_t flag;
+
+        /// User-presentable short name of the feature flag.
+        const wchar_t *name;
```

Comment:
Any reason these are wchar_t instead of wcstring?

Reply:
There will be string literals, so it's a little cheaper to initialize since it can be in a read-only segment, and doesn't need malloc().